gnd/src/main/java/com/google/android/gnd/MainViewModel.java

```diff
@@ -151,7 +173,16 @@ void onSignedOut() {
 
   private void onSignedIn() {
     hideProgressDialog();
-    navigator.navigate(HomeScreenFragmentDirections.showHomeScreen());
+    if (termsState.getValue() == LoadState.NOT_FOUND) {
```

Comment:
There's a race condition here: if the terms aren't loaded before the sign in completes `termsState.getValue()` will return null, and the user will be directed to the home screen. I think the right thing to do here would be to join into the stream that triggers onSignInState so that it doesn't call the observer until the terms are loaded.